#731244 - Add NFSROOT_SERVER option to explicitly set NFS server

ekoyle opened the pull request:
Add a new optional configuration option to /etc/fai/nfsroot.conf called NFSROOT_SERVER which allows users to set the server for NFSROOT explicitly.

This is needed if the DHCP server is different from the FAI/NFS server because dracut prefers the DHCP Server Identifier (option 54, which contains the DHCP server's IPv4 address -- see RFC2132 section 9.7) over any other DHCP option.

This will resolve [731244 in Debian's BTS](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=731244).

Mrfai commented:
Are you sure you can use the servername here? The dracut.cmdline man page says it needs the IP address after root=
We should also add a comment, that this variable is only used by fai-chboot.

ekoyle commented:
I'm fairly sure that the server name will work if name resolution is configured properly, but I have removed it to match the dracut docs.  I also added a comment on why this is needed.

Mrfai commented:
I have now added your patch. Thanks a lot.
